miniapp/miniapp_tridiag_solver.cpp

```diff
@@ -0,0 +1,197 @@
+using dlaf::matrix::Distribution;
+using dlaf::matrix::MatrixMirror;
+
+struct Options
+    : dlaf::miniapp::MiniappOptions<dlaf::miniapp::SupportReal::Yes, dlaf::miniapp::SupportComplex::Yes> {
+  SizeType m;
+  SizeType mb;
+
+  Options(const pika::program_options::variables_map& vm)
+      : MiniappOptions(vm), m(vm["matrix-size"].as<SizeType>()), mb(vm["block-size"].as<SizeType>()) {
+    DLAF_ASSERT(m > 0, m);
+    DLAF_ASSERT(mb > 0, mb);
+
+    if (do_check != dlaf::miniapp::CheckIterFreq::None) {
+      std::cerr << "Warning! At the moment result checking it is not implemented." << std::endl;
+      do_check = dlaf::miniapp::CheckIterFreq::None;
+    }
+  }
+
+  Options(Options&&) = default;
+  Options(const Options&) = default;
+  Options& operator=(Options&&) = default;
+  Options& operator=(const Options&) = default;
+};
+}
+
+struct TridiagSolverMiniapp {
+  template <Backend backend, typename T>
+  static void run(const Options& opts) {
+    using RealParam = dlaf::BaseType<T>;
+
+    Communicator world(MPI_COMM_WORLD);
+    CommunicatorGrid comm_grid(world, opts.grid_rows, opts.grid_cols, Ordering::ColumnMajor);
+
+    // Allocate the tridiagonl, eigenvalues and eigenvectors matrices
+    const Distribution dist_trd(LocalElementSize(opts.m, 2), TileElementSize(opts.mb, 2));
+    const Distribution dist_evals(LocalElementSize(opts.m, 1), TileElementSize(opts.mb, 1));
+    const Distribution dist_evecs(GlobalElementSize(opts.m, opts.m), TileElementSize(opts.mb, opts.mb),
+                                  comm_grid.size(), comm_grid.rank(), {0, 0});
+
+    Matrix<RealParam, Device::CPU> tridiag(dist_trd);
+    Matrix<RealParam, Device::CPU> evals(dist_evals);
+    Matrix<T, Device::CPU> evecs(dist_evecs);
+
+    // Initialize a random symmetric tridiagonal matrix using two arrays for the diagonal and the
+    // off-diagonal. The random numbers are in the range [-1, 1].
+    //
+    // Note: set_random() is not used here because the two arrays can be more easily reused to initialize
+    //       the same tridiagonal matrix but with explicit zeros for correctness checking further down.
+    const auto initRandomArray = [size = dlaf::to_sizet(opts.m)](const SizeType seed) {
+      std::vector<RealParam> diag_arr(size);
+      dlaf::matrix::util::internal::getter_random<RealParam> diag_rand_gen(seed);
+      std::generate(std::begin(diag_arr), std::end(diag_arr), diag_rand_gen);
+      return diag_arr;
+    };
+    const SizeType diag_seed = opts.m;
+    const SizeType offdiag_seed = opts.m + 1;
+    const std::vector<RealParam> diag_arr = initRandomArray(diag_seed);
+    const std::vector<RealParam> offdiag_arr = initRandomArray(offdiag_seed);
+
+    auto tridagReference = [&diag_arr, &offdiag_arr](GlobalElementIndex i) {
+      const size_t row_index = dlaf::to_sizet(i.row());
+      if (i.col() == 0) {
+        return diag_arr[row_index];
+      }
+      else {
+        return offdiag_arr[row_index];
+      }
+    };
```

Comment:
Isn't it better to create tridiag_reference as a (m x 2) random matrix?

Reply:
see https://github.com/eth-cscs/DLA-Future/pull/753/commits/7761e7988caeaa02330bf8c7f7ebc49645272a08